django_mongodb/expressions.py

```diff
@@ -79,8 +92,63 @@ def order_by(self, compiler, connection):
     return self.expression.as_mql(compiler, connection)
 
 
-def query(self, compiler, connection):  # noqa: ARG001
-    raise NotSupportedError("Using a QuerySet in annotate() is not supported on MongoDB.")
+def query(self, compiler, connection, lookup_name=None):
+    subquery_compiler = self.get_compiler(connection=connection)
+    subquery_compiler.pre_sql_setup(with_col_aliases=False)
+    columns = subquery_compiler.get_columns()
+    field_name, expr = columns[0]
+    subquery = subquery_compiler.build_query(
+        columns
+        if subquery_compiler.query.annotations or not subquery_compiler.query.default_cols
+        else None
+    )
+    table_output = f"__subquery{len(compiler.subqueries)}"
+    from_table = next(
+        e.table_name for alias, e in self.alias_map.items() if self.alias_refcount[alias]
+    )
+    subquery.lookup_data = {
```

Comment:
The name lookup_data does not readily indicate its purpose and I don't see any explanatory comments about it.